C/ds/inorder_predecessor_successor.c

```diff
@@ -0,0 +1,241 @@
+            case 3:
+            {
+                if(root==NULL)
+                printf("Tree is empty\n");
+                else
+                {
+                    printf("Enter the node for which you want inorder predecessor: ");
+                    scanf("%d",&n);
+                    in_pre(root,n);
+                }
+                break;
+            }
+            case 4:
+            {
+                if(root==NULL)
+                printf("Tree is empty\n");
+                else
+                {
+                    printf("Enter the node for which you want inorder successor: ");
+                    scanf("%d",&n);
+                    in_succ(root,n);
+                }
+                break;
+            }
+            default: printf("Not a choice\n");
+            break;
+        }
+        printf("Do you want to continue:(Y/y) ");
+        scanf(" %c",&c);
+    } while (c=='Y' || c=='y');
+    return 0;
+}
+/*
+Menu:
+1.Insert                                2.Inorder display
+3.Inorder Predecessor of a node         4.Inorder Successor of a node
+Enter your choice: 1
+Enter data: 14
+Do you want to continue:(Y/y) y
+
+Menu:
+1.Insert                                2.Inorder display
+3.Inorder Predecessor of a node         4.Inorder Successor of a node
+Enter your choice: 1
+Enter data: 9
+Do you want to continue:(Y/y) y
+
+Menu:
+1.Insert                                2.Inorder display
+3.Inorder Predecessor of a node         4.Inorder Successor of a node
+Enter your choice: 1
+Enter data: 3
+Do you want to continue:(Y/y) y
+
+Menu:
+1.Insert                                2.Inorder display
+3.Inorder Predecessor of a node         4.Inorder Successor of a node
+Enter your choice: 1
+Enter data: 5
+Do you want to continue:(Y/y) y
+
+Menu:
+1.Insert                                2.Inorder display
+3.Inorder Predecessor of a node         4.Inorder Successor of a node
+Enter your choice: 1 
+Enter data: 18
+Do you want to continue:(Y/y) y
+
+Menu:
+1.Insert                                2.Inorder display
+3.Inorder Predecessor of a node         4.Inorder Successor of a node
+Enter your choice: 1
+Enter data: 15
+Do you want to continue:(Y/y) y
+
+Menu:
+1.Insert                                2.Inorder display
+3.Inorder Predecessor of a node         4.Inorder Successor of a node
+Enter your choice: 2
+In-order Display: 3  5  9  14  15  18
+Do you want to continue:(Y/y) y
+
+Menu:
+1.Insert                                2.Inorder display
+3.Inorder Predecessor of a node         4.Inorder Successor of a node
+Enter your choice: 3
+Enter the node for which you want inorder predecessor: 5
+Predecessor of 5 is 3
+Do you want to continue:(Y/y) y
+
+Menu:
+1.Insert                                2.Inorder display
+3.Inorder Predecessor of a node         4.Inorder Successor of a node
+Enter your choice: 4
+Enter the node for which you want inorder successor: 5
+Successor of 5 is 9
+Do you want to continue:(Y/y) n
+*/
```

Comment:
- Remove the extra test case 3  5  9  14 (Use only four values as sample input)
- Add Time and Space Complexity at the end of the Code